src/commands/search.rs

```diff
@@ -30,6 +35,10 @@ fn run_search(directory: &Path, query: &str) -> tantivy::Result<()> {
         })
         .map(|(field, _)| field)
         .collect();
+    let filter_fields: BTreeSet<u32> = schema
+        .fields()
+        .filter(|&(_, ref field_entry)| filter_fields_str.contains(field_entry.name()))
```

Comment:
Iterating over the filter_fields and callign schema.get_field(...) would greatly simplify this code.